### PR TITLE
Added fast parse options feature. Changed options interface.

### PR DESCRIPTION
**Pull request by Moliholy**
Also added a markdown file in /doc explaining the problems we found out while running with IgProf

### CODE REVIEW

File: cvmfs/loader.cc [hunk 1]
```diff
@@ -55,6 +55,7 @@ struct CvmfsOptions {
   int grab_mountpoint;
   int cvmfs_suid;
   int disable_watchdog;
+  int fast_parse;
```
Comment (reneme) [inline cvmfs/loader.cc]: Maybe clarify in the name of that variable that you deal with the option parsing? `fast_option_parsing` or `minimal_option_parsing`?

**Comment by jblomer** [comment]
Many thanks, that looks already quite good!  

Since this is your first larger commit, we'll probably take it through a few rounds before merging.  I'd like to ask you to create a feature branch or a bugfix branch for every pull request you're working on.  Also, please create separate pull request for
  - The changes to .gitignore
  - The igprof document
  - The interface change to options and the unit test
  - The additional logic for fast_parse (which we should probably rename to simple_parse)

We'll then work on the details of the individual pull requests.

File: cvmfs/talk.cc [hunk 2]
```diff
@@ -510,11 +511,12 @@ static void *MainTalk(void *data __attribute__((unused))) {
 /**
  * Init the socket.
  */
-bool Init(const string &cachedir) {
+bool Init(const string &cachedir, OptionsManager *optionsManager) {
```
Comment (Moliholy) [inline cvmfs/talk.cc]: I forgot to change the variable name to something like options_manager, according to the Google C++ style guide.

**Comment by jblomer** [comment]
As discussed, closing this one in lieu for the four separate ones.

**Pull request closed without merging**